internal/preflight/preflight.go

```diff
@@ -0,0 +1,67 @@
+package preflight
+
+import (
+	"github.com/kubernetes-sigs/kernel-module-management/api/v1beta2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const VerificationStatusReasonVerified = "Verification successful (%s), this Module will not be verified again in this Preflight CR"
+
+//go:generate mockgen -source=preflight.go -package=preflight -destination=mock_preflight_api.go
+
+type PreflightAPI interface {
+	SetModuleStatus(pv *v1beta2.PreflightValidation, namespace, name, status, reason string)
+	GetModuleStatus(pv *v1beta2.PreflightValidation, namespace, name string) string
+	AllModulesVerified(pv *v1beta2.PreflightValidation) bool
+}
+
+func NewPreflightAPI() PreflightAPI {
+	return &preflight{}
+}
+
+type preflight struct {
+}
+
+func (p *preflight) SetModuleStatus(pv *v1beta2.PreflightValidation, namespace, name, status, reason string) {
+	stage := v1beta2.VerificationStageImage
+	if status == v1beta2.VerificationSuccess || status == v1beta2.VerificationFailure {
+		stage = v1beta2.VerificationStageDone
+	}
+	newStatus := v1beta2.PreflightValidationModuleStatus{
+		Name:      name,
+		Namespace: namespace,
+		CRBaseStatus: v1beta2.CRBaseStatus{
+			VerificationStatus: status,
+			StatusReason:       reason,
+			VerificationStage:  stage,
+			LastTransitionTime: metav1.Now(),
+		},
+	}
+	for i, moduleStatus := range pv.Status.Modules {
+		if moduleStatus.Name == name && moduleStatus.Namespace == namespace {
+			pv.Status.Modules[i] = newStatus
+			return
+		}
+	}
+
+	pv.Status.Modules = append(pv.Status.Modules, newStatus)
+}
+
+func (p *preflight) GetModuleStatus(pv *v1beta2.PreflightValidation, namespace, name string) string {
+	for _, moduleStatus := range pv.Status.Modules {
+		if moduleStatus.Name == name && moduleStatus.Namespace == namespace {
+			return moduleStatus.VerificationStatus
+		}
+	}
+	return ""
```

Comment:
Should we return an error here instead of an empty value? errors are more often checked than empty values.

Reply:
this is not an error, the module is not present, since it might be that we have not yet pick it up. The empty status is ok in this case